cunumeric/array.py

```diff
@@ -1252,6 +1252,92 @@ def dump(self, file):
     def dumps(self):
         return self.__array__().dumps()
 
+    def fft(self, s, axes, kind, direction, norm):
+        # Dimensions check
+        if self.ndim > 3:
+            raise NotImplementedError(
+                f"{self.ndim}-D arrays are not supported yet"
+            )
+
+        # Type
+        fft_output_type = self.dtype
```

Comment:
General comment. For future proofing a bit (e.g. half/bfloat16 support), would it be worth factoring out the `precision` (`single`, `double`, `half`, `bfloat`) from the `type` (`R2C`/`C2R`/`C2C`)? Then we can convert `{precision, type}` to `CUFFT_C2C, CUFFT_Z2D, ...` as needed right before calling the `cuFFT` APIs?

Reply:
What about using `cufftXtMakePlanMany` with https://docs.nvidia.com/cuda/cufft/index.html#common-cudadatatype-types. I agree it's a bit of a PITA for right now.

Reply:
For future release.